File: Math/codeforces_1931D.py

# Problem link ===>> https://codeforces.com/problemset/problem/1931/D?locale=en

def solve():
    n, x, y = map(int, input().split())
    a = list(map(int, input().split()))

    ans = 0
    mp = {}
    for i in range(n):
        rem1 = a[i] % y
        rem2 = (x - a[i] % x) % x

        ans += mp.get(rem1, {}).get(rem2, 0)
        if rem1 not in mp:
            mp[rem1] = {}
        mp[rem1][a[i] % x] = mp[rem1].get(a[i] % x, 0) + 1

    print(ans)


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        solve()

File: Math/codeforces_1931D.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1931/D?locale=en
// submission link ===>> https://codeforces.com/contest/1931/submission/247519574

#include <bits/stdc++.h>
using namespace std;

using ll = long long;

void solve()
{
    int n, x, y;
    cin >> n >> x >> y;
    vector<int> a(n);
    for (auto &x : a)
        cin >> x;

    ll ans = 0;
    map<int, map<int, int>> mp;
    for (int i = 0; i < n; ++i)
    {
        ans += mp[a[i] % y][(x - a[i] % x) % x];
        mp[a[i] % y][a[i] % x]++;
    }
    cout << ans << '\n';
}

int main(void)
{
    int t;
    cin >> t;
    for (int i = 0; i < t; ++i)
    {
        solve();
    }
    return 0;
}